src/structs/jpeg_header.rs

```diff
@@ -295,6 +298,36 @@ pub struct JPegHeader {
     pub cs_sal: u8,  // successive approximation bit pos low
 }
 
+pub struct JPegEncodingInfo {
+    pub jpeg_header: JPegHeader,
+    pub truncate_components: TruncateComponents,
+
+    /// A list containing one entry for each scan segment.  Each entry contains the number of restart intervals
+    /// within the corresponding scan segment.
+    pub rst_cnt: Vec<i32>,
+
+    /// the mask for padding out the bitstream when we get to the end of a reset block
+    pub pad_bit: Option<u8>,
+
+    pub rst_cnt_set: bool,
+
+    /// count of scans encountered so far
+    pub scnc: usize,
+}
```

Comment:
What is the idea behind duplication of data inside this structure?

Reply:
Avoids jpeg_writer having a dependency on LeptonHeader and ThreadHandoff, also because now we clone the data structures instead of passing by reference, so I wanted to avoid having to clone the entire LeptonHeader to pass it to each writing threads.